docs/Basic Concepts.md

```diff
@@ -0,0 +1,61 @@
+# Cloudflow - Basic Concepts
+
+This document gives you an overview of the main building blocks of a cloudflow application.
+
+## Application
+
+A Cloudflow application is a self-contained distributed system of data processing services connected together by data streams. A Cloudflow cluster can host multiple applications.
+
+![Application](images/apps-1.png?resize=400,400&classes=center)
+
+A Cloudflow application consists of the following basic building blocks:
+
+1. Streamlets
+2. Blueprint
+3. Deployed
+
+We will go through each of the above in the following sections.
+
+![Application Building Blocks](images/apps-2.png?raw=true "Application Building Blocks")
+
+
+## Streamlets
+
+A streamlet is one of the building blocks of a Cloudflow application. Each streamlet represents a stage in the data transformation pipeline. Streamlets can have inlets and outlets that provide the conduit for the flow of data. Each inlet and outlet is typed and can handle data _of that specific type_ only.
```

Comment:
s"...data transformation pipeline"/".... data transformation graph"?

Reply:
@skonto strictly speaking it's not a graph - it cannot have a cycle .. we can say `tree` but I think `pipelines` is more intuitive